src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+    }
+
+    // casts req
+    this->actor.shape.rot.x = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.x;
+    this->actor.shape.rot.y = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.y;
+    this->actor.shape.rot.z = ((s16)randPlusMinusPoint5Scaled(5000.0f)) + this->actor.world.rot.z;
+    if (this->niwType == ENNIW_TYPE_REGULAR) {
+        if (Actor_HasNoParent(&this->actor, globalCtx)) {
+            this->actor.shape.rot.z = 0;
+            rotZ = this->actor.shape.rot.z;
+            this->unknownState28E = 5;
+            this->actor.flags |= 0x1; // targetable ON
+            this->actionFunc = EnNiw_Thrown;
+            this->actor.shape.rot.y = rotZ;
+            this->actor.shape.rot.x = rotZ;
+        }
+
+    } else if (this->unk2BC.z != 0.0f) {
+        // if not regular..? when does this go off?
+        this->actor.shape.rot.z = 0;
+        rotZ = this->actor.shape.rot.z;
+        this->actor.velocity.y = 8.0f;
+        this->actor.speedXZ = 4.0f;
+        this->actor.gravity = -2.0f;
+        this->unknownState28E = 5;
+        this->unk2EC = 0;
+        this->niwType = ENNIW_TYPE_REGULAR;
+        this->actor.shape.rot.y = rotZ;
+        this->actor.shape.rot.x = rotZ;
+         Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor,  &sCylinderInit );
+        Math_Vec3f_Copy(&this->unk2BC, &vec3fcopy);
+        this->actor.flags |= 0x1; // targetable ON
+        this->actionFunc = EnNiw_Thrown;
+    }
+    func_80891320(this, globalCtx, 2);
+}
+
+// action function: recently thrown, and also hopping on the floor
+void EnNiw_Thrown(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->unk2EC == 0) {
+        if ((this->actor.bgCheckFlags & 1)) { // about to hit the floor
+            this->unk2EC = 1;
+            this->unkTimer252 = 80; // hop timer
+            this->actor.speedXZ = 0.0f;
+            this->actor.velocity.y = 4.0f;
+        } else {
+            return; // wait until back on floor
+        }
+    } else {
+        if ((this->actor.bgCheckFlags & 1)) { // about to hit the floor
+            this->sfxTimer1 = 0;
+            this->actor.velocity.y = 4.0f; // vertical hop
+            this->unk29E = 1;
+        }
+        if (this->unkTimer252 == 0) {
+            this->unkTimer254 = 100;
+            this->unkTimer250 = 0;
+            this->unk2EC = 0;
+            EnNiw_SetupRunning(this);
+            return;
+        }
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        // picked up again before could run off
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M); // crow
+        this->sfxTimer1 = 30;
+        this->unk2EC = 0;
+        this->unkTimer250 = 30;
+        this->actor.flags &= ~0x1; // targetable OFF
+        this->unknownState28E = 4;
+        this->actionFunc = EnNiw_Held;
+        this->actor.speedXZ = 0.0f;
+    } else {
+        if (this->unkTimer252 >= 6) {
```

Comment:
```suggestion
        if (this->unkTimer252 > 5) {
```
There's an instance above where this timer is set to 5.